packages/medusa/src/loaders/helpers/resolve-plugins.ts

```diff
@@ -1,148 +1,81 @@
-import { ConfigModule, PluginDetails } from "@medusajs/framework/types"
-import { isString } from "@medusajs/framework/utils"
 import fs from "fs"
-import { sync as existsSync } from "fs-exists-cached"
-import path, { isAbsolute } from "path"
+import path from "path"
+import { isString, readDir } from "@medusajs/framework/utils"
+import { ConfigModule, PluginDetails } from "@medusajs/framework/types"
 
 export const MEDUSA_PROJECT_NAME = "project-plugin"
+const MEDUSA_APP_SOURCE_PATH = "src"
+
 function createPluginId(name: string): string {
   return name
 }
 
-function createFileContentHash(path, files): string {
+function createFileContentHash(path: string, files: string): string {
   return path + files
 }
 
-function getExtensionDirectoryPath() {
-  return "src"
-}
-
-/**
- * Load plugin details from a path. Return undefined if does not contains a package.json
- * @param pluginName
- * @param path
- * @param includeExtensionDirectoryPath should include src | dist for the resolved details
- */
-function loadPluginDetails({
-  pluginName,
-  resolvedPath,
-  includeExtensionDirectoryPath,
-}: {
-  pluginName: string
-  resolvedPath: string
-  includeExtensionDirectoryPath?: boolean
-}) {
-  if (existsSync(`${resolvedPath}/package.json`)) {
-    const packageJSON = JSON.parse(
-      fs.readFileSync(`${resolvedPath}/package.json`, `utf-8`)
-    )
-    const name = packageJSON.name || pluginName
-
-    const extensionDirectoryPath = getExtensionDirectoryPath()
-    const resolve = includeExtensionDirectoryPath
-      ? path.join(resolvedPath, extensionDirectoryPath)
-      : resolvedPath
-
-    return {
-      resolve,
-      name,
-      id: createPluginId(name),
-      options: {},
-      version: packageJSON.version || createFileContentHash(path, `**`),
-    }
-  }
-
-  // Make package.json a requirement for local plugins too
-  throw new Error(`Plugin ${pluginName} requires a package.json file`)
-}
-
 /**
  * Finds the correct path for the plugin. If it is a local plugin it will be
  * found in the plugins folder. Otherwise we will look for the plugin in the
  * installed npm packages.
- * @param {string} pluginName - the name of the plugin to find. Should match
+ * @param {string} pluginPath - the name of the plugin to find. Should match
  *    the name of the folder where the plugin is contained.
  * @return {object} the plugin details
  */
-function resolvePlugin(pluginName: string): {
-  resolve: string
-  id: string
-  name: string
-  options: Record<string, unknown>
-  version: string
-} {
-  if (!isAbsolute(pluginName)) {
-    let resolvedPath = path.resolve(`./plugins/${pluginName}`)
```

Comment:
When using `require.resolve`, we shouldn't worry about absolute or relative paths separately.